File: trunk/CreaObjetos/CreaObjetos/Formas/frmgeneraOpciones.vb
Imports System.IO
Public Class frmgeneraOpciones

    Private rdbOpciones(3) As RadioButton
    Private opcionSelecta As Integer
    Private _ManejaClases As List(Of ManejaClase)
    Private directorio As String
    Public Property manejaClases() As List(Of ManejaClase)
        Get
            Return Me._ManejaClases
        End Get
        Set(ByVal value As List(Of ManejaClase))
            Me._ManejaClases = value
        End Set
    End Property

    Private Sub btnGeneraClase_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnGeneraClase.Click
        Dim strCodigo As String
        Dim neimspace As String
        neimspace = ""
        While (neimspace.Equals(""))
            neimspace = InputBox("Que namespace deseas utilizar")
        End While

        'strCodigo = "imports system.data.Sqlclient" & vbCrLf
        'strCodigo &= "namespace " & neimspace & vbCrLf
        strCodigo = Me.procesaClases(neimspace) & vbCrLf
        'strCodigo &= "end namespace "

        Select Case opcionSelecta
            Case 0
                Clipboard.SetDataObject(strCodigo)
            Case 1
                Me.TextBox1.Text = strCodigo
            Case 2
                MessageBox.Show("Los archivos de clases han sido creados")
            Case 3
                Me.SaveTextToFile(strCodigo, Me.buscaDirectorio(), Me.nombraArchivo, "Jodeos")
                MessageBox.Show("El archivo de clases ha sido creado")
        End Select
    End Sub

    Private Sub ctlGeneraOpciones_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

        Dim i As Integer
        Me.rdbOpciones(0) = Me.rdbClipboard
        Me.rdbOpciones(1) = Me.rdbTexto
        Me.rdbOpciones(2) = Me.rdbVarios
        Me.rdbOpciones(3) = Me.rdbArchivo

        For i = 0 To Me.rdbOpciones.Length - 1
            Me.rdbOpciones(i).Tag = New String("" & i)
            AddHandler Me.rdbOpciones(i).CheckedChanged, AddressOf escogeOpcion
        Next
        Me.rdbOpciones(0).Checked = True

    End Sub

    Private Sub escogeOpcion(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Me.opcionSelecta = CType(sender.Tag, Int32)
    End Sub

    Public Function procesaClases(ByVal spacename As String) As String
        Dim c As Integer
        Dim sclases As String
        Dim claces As New List(Of Clase)

        sclases = ""

        If Me.rdbVarios.Checked Then
            directorio = Me.buscaDirectorio
        End If

        For c = 0 To Me.manejaClases.Count - 1
            Dim mnjClsTemp As ManejaClase = Me.manejaClases(c)
            'Checa los atributos de las clases y agrega los checados

            If mnjClsTemp.chbIncluir.Checked Then
                Dim clastemp As New Clase

                'Inicializa propiedades de clase
                clastemp.llave = mnjClsTemp.Clase.llave
                clastemp.NombreTabla = mnjClsTemp.Clase.NombreTabla

                'inicializa Lista de campos
                clastemp.Campos = New List(Of Propiedad)
                Dim indexChecked As Integer
                For Each indexChecked In mnjClsTemp.clbCampos.CheckedIndices
                    clastemp.Campos.Add(mnjClsTemp.Clase.Campos(indexChecked))
                Next

                Dim genera As New GeneraClase
                Dim gnrMnj As New generaManejador
                Dim gnrDsn As New generaDiseno
                Dim gnrFrm As New generaForma

                'Diseño
                gnrDsn.neimespeis = spacename
                gnrDsn.clace = clastemp
                'Forma
                gnrFrm.neimespeis = spacename
                gnrFrm.clace = clastemp
                ''Manejador
                gnrMnj.neimespeis = spacename
                gnrMnj.clace = clastemp
                ''Clase

                genera.clase = clastemp
                genera.neimespeis = spacename

                sclases &= genera.GeneraClase()
                If Me.rdbVarios.Checked Then
                    Me.SaveTextToFile(gnrFrm.generaForma, directorio, "Frm" & clastemp.NombreTabla, "Jodeos")
                    Me.SaveTextToFile(gnrDsn.generaDiseno, directorio, "Frm" & clastemp.NombreTabla & ".designer", "Jodeos")
                    Me.SaveTextToFile(gnrMnj.generaManeja, directorio, "maneja" & clastemp.NombreTabla, "Jodeos")
                    Me.SaveTextToFile(genera.GeneraClase(), directorio, clastemp.nombre, "Jodeos")
                End If

            End If

        Next

        Return sclases
    End Function

    Public Function SaveTextToFile(ByVal strData As String, _
    ByVal FullPath As String, _
    ByVal nombreArchivo As String, _
 Optional ByVal ErrInfo As String = "") As Boolean

        Dim bAns As Boolean = False

        Dim objReader As StreamWriter
        Try
            objReader = New StreamWriter(FullPath & "\" & nombreArchivo & ".vb")
            objReader.Write(strData)
            objReader.Close()
            bAns = True
        Catch Ex As Exception
            ErrInfo = Ex.Message
        End Try
        Return bAns
    End Function
    Public Function buscaDirectorio() As String
        Dim fbdCodigo As New FolderBrowserDialog
        fbdCodigo.RootFolder = Environment.SpecialFolder.Desktop
        ' Select the C:\Windows directory on entry.
        fbdCodigo.SelectedPath = "c:\windows"
        ' Prompt the user with a custom message.
        fbdCodigo.Description = "Select the source directory"
        fbdCodigo.ShowDialog()

        Return fbdCodigo.SelectedPath
    End Function
    Public Function nombraArchivo() As String
        Dim sarchivo As New String("")
        While (sarchivo.Equals(""))
            sarchivo = InputBox("Como quiero que se llame el archivo de Salida")
        End While
        Return sarchivo
    End Function
End Class

File: CreaObjetos/CreaObjetos.ConvertedToC#/Formas/frmgeneraOpciones.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.IO;
using WindowsApplication1.GeneraCodigo;

namespace WindowsApplication1
{
	public partial class frmgeneraOpciones
	{

		private RadioButton[] rdbOpciones = new RadioButton[4];
		private int opcionSelecta;
		private List<ManejaClase> _ManejaClases;
		private string directorio;
		public List<ManejaClase> manejaClases {
			get { return this._ManejaClases; }
			set { this._ManejaClases = value; }
		}

		private void btnGeneraClase_Click(System.Object sender, System.EventArgs e)
		{
			string strCodigo = null;
			string neimspace = null;
			neimspace = "";
			while ((neimspace.Equals(""))) {
				neimspace = Interaction.InputBox("Que namespace deseas utilizar");
			}

			//strCodigo = "imports system.data.Sqlclient" & vbCrLf
			//strCodigo &= "namespace " & neimspace & vbCrLf
			strCodigo = this.procesaClases(neimspace) + Constants.vbCrLf;
			//strCodigo &= "end namespace "

			switch (opcionSelecta) {
				case 0:
					Clipboard.SetDataObject(strCodigo);
					break;
				case 1:
					this.TextBox1.Text = strCodigo;
					break;
				case 2:
					MessageBox.Show("Los archivos de clases han sido creados");
					break;
				case 3:
					this.SaveTextToFile(strCodigo, this.buscaDirectorio(), this.nombraArchivo(), "Jodeos");
					MessageBox.Show("El archivo de clases ha sido creado");
					break;
			}
		}


		private void ctlGeneraOpciones_Load(object sender, System.EventArgs e)
		{
			int i = 0;
			this.rdbOpciones[0] = this.rdbClipboard;
			this.rdbOpciones[1] = this.rdbTexto;
			this.rdbOpciones[2] = this.rdbVarios;
			this.rdbOpciones[3] = this.rdbArchivo;

			for (i = 0; i <= this.rdbOpciones.Length - 1; i++) {
				this.rdbOpciones[i].Tag =i.ToString();
				this.rdbOpciones[i].CheckedChanged += escogeOpcion;
			}
			this.rdbOpciones[0].Checked = true;

		}

		private void escogeOpcion(System.Object sender, System.EventArgs e)
		{
			this.opcionSelecta = (int)((RadioButton)(sender)).Tag;
		}

		public string procesaClases(string spacename)
		{
			int c = 0;
			string sclases = null;
			List<Clase> claces = new List<Clase>();

			sclases = "";

			if (this.rdbVarios.Checked) {
				directorio = this.buscaDirectorio();
			}

			for (c = 0; c <= this.manejaClases.Count - 1; c++) {
				ManejaClase mnjClsTemp = this.manejaClases[c];
				//Checa los atributos de las clases y agrega los checados

				if (mnjClsTemp.chbIncluir.Checked) {
					Clase clastemp = new Clase();

					//Inicializa propiedades de clase
					clastemp.llave = mnjClsTemp.Clase.llave;
					clastemp.NombreTabla = mnjClsTemp.Clase.NombreTabla;

					//inicializa Lista de campos
					clastemp.Campos = new List<Propiedad>();
					int indexChecked = 0;
					foreach (int indexChecked_loopVariable in mnjClsTemp.clbCampos.CheckedIndices) {
						indexChecked = indexChecked_loopVariable;
						clastemp.Campos.Add(mnjClsTemp.Clase.Campos[indexChecked]);
					}




					if (this.rdbVarios.Checked) {
						if (this.rdbWapl.Checked) {
							sclases += this.VariosArchivosAppl(spacename, clastemp);
						} else if (this.rdbASP.Checked) {
							sclases += this.VariosArchivosASP(spacename, clastemp);
						}

					}

				}

			}

			return sclases;
		}
		public string VariosArchivosASP(string spacename, Clase clastemp)
		{
			string sclases = string.Empty;

			if (this.chkClasesASP.Checked) {
				var gnrDsn = new creaPagina();
				generaCodPagina gnrCdPg = new generaCodPagina();
				//Diseño
				gnrDsn.neimespeis = spacename;
				gnrDsn.clace = clastemp;
				//Pagina
				gnrCdPg.neimespeis = spacename;
				gnrCdPg.clace = clastemp;
				this.SaveTextToFile(gnrDsn.GeneraPagina(), directorio + "\\Views", clastemp.NombreTabla + "View.aspx", "Jodeos");
				this.SaveTextToFile(gnrCdPg.generaForma(), directorio + "\\Views", clastemp.NombreTabla + "View.aspx.vb", "Jodeos");
			}

			if (this.chkManejador.Checked) {
				generaCodPagina gnrCdPg = new generaCodPagina();
				//Pagina
				gnrCdPg.neimespeis = spacename;
				gnrCdPg.clace = clastemp;
				this.SaveTextToFile(gnrCdPg.generaForma(), directorio + "\\Views", clastemp.NombreTabla + "View.aspx.vb", "Jodeos");
			}


			if (this.chkClases.Checked) {
				GeneraClaseASP genera = new GeneraClaseASP();

				genera.clase = clastemp;
				genera.neimespeis = spacename;
				sclases += genera.GeneraClase();

				this.SaveTextToFile(genera.GeneraClase(), directorio + "\\Models", clastemp.nombre() + ".vb", "Jodeos");
			}
			return sclases;
		}
		public string VariosArchivosAppl(string spacename, Clase clastemp)
		{
			string sclases = string.Empty;
			if (this.chkInterfaz.Checked) {
				var gnrDsn = new creaDiseno();
				var gnrFrm = new generaForma();
				//Diseño
				gnrDsn.neimespeis = spacename;
				gnrDsn.clace = clastemp;
				//Forma
				gnrFrm.neimespeis = spacename;
				gnrFrm.clace = clastemp;
				this.SaveTextToFile(gnrFrm.creaForma(), directorio + "/Views/", "Frm" + clastemp.NombreTabla, "Jodeos");
				this.SaveTextToFile(gnrDsn.GeneraDiseno(), directorio + "/Views/", "Frm" + clastemp.NombreTabla + ".designer", "Jodeos");
			}

			if (this.chkManejador.Checked) {
				generaManejador gnrMnj = new generaManejador();
				//'Manejador
				gnrMnj.neimespeis = spacename;
				gnrMnj.clace = clastemp;
				//'Clase
				this.SaveTextToFile(gnrMnj.generaManeja(), directorio + "/Actions/", "maneja" + clastemp.NombreTabla + ".vb", "Jodeos");
			}


			if (this.chkClases.Checked) {
				GeneraClase genera = new GeneraClase();
				genera.Clase = clastemp;
				genera.Neimespeis = spacename;
				sclases += genera.creaClase();
				this.SaveTextToFile(genera.creaClase(), directorio + "/Models/", clastemp.nombre() + ".vb", "Jodeos");
			}
			return sclases;
		}

		public bool SaveTextToFile(string strData, string FullPath, string nombreArchivo, string ErrInfo = "")
		{

			if (!WindowsApplication1.My.MyProject.Computer.FileSystem.DirectoryExists(FullPath)) {
				WindowsApplication1.My.MyProject.Computer.FileSystem.CreateDirectory(FullPath);
			}
			bool bAns = false;

			StreamWriter objReader = null;
			try {
				objReader = new StreamWriter(FullPath + "\\" + nombreArchivo);
				objReader.Write(strData);
				objReader.Close();
				bAns = true;
			} catch (Exception Ex) {
				ErrInfo = Ex.Message;
			}
			return bAns;
		}
		public string buscaDirectorio()
		{
			FolderBrowserDialog fbdCodigo = new FolderBrowserDialog();
			fbdCodigo.RootFolder = Environment.SpecialFolder.Desktop;
			// Select the C:\Windows directory on entry.
			fbdCodigo.SelectedPath = "c:\\";
			// Prompt the user with a custom message.
			fbdCodigo.Description = "Select the source directory";
			fbdCodigo.ShowDialog();

			return fbdCodigo.SelectedPath;
		}
		public string nombraArchivo()
		{
			string sarchivo = string.Empty;
			while ((sarchivo.Equals(""))) {
				sarchivo = Interaction.InputBox("Como quiero que se llame el archivo de Salida");
			}
			return sarchivo;
		}



		private void rdbWapl_CheckedChanged(System.Object sender, System.EventArgs e)
		{
			this.ActivaChkCodigo(this.rdbWapl.Checked);
		}


		private void ActivaChkCodigo(bool Activado)
		{
			this.chkPagina.Visible = !Activado;
			this.chkClasesASP.Visible = !Activado;
			this.chkMngASP.Visible = !Activado;

			this.chkClases.Visible = Activado;
			this.chkInterfaz.Visible = Activado;
			this.chkManejador.Visible = Activado;

		}
		public frmgeneraOpciones()
		{
			Load += ctlGeneraOpciones_Load;
			InitializeComponent();
		}
	}
}


